File: C3_WoWFeedReader/FeedGrabberCS/WowFeedGrabber/WowFeedGrabber/HtmlConvert.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace WowFeedGrabber 
{
    public static class HtmlConvert 
    {
        private static readonly Dictionary<string, string> characterPlaceholders = new Dictionary<string, string>()
                                                                            {
                                                                                { "&amp;", "&" },
                                                                                { "&lt;", "<" },
                                                                                { "&gt;", ">" },
                                                                                { "&quot;", "'" },
                                                                                { "&apos;", "'" }
                                                                            };
        public static string ToPlainText(string html)
        {
            if (html == null)
                throw new ArgumentNullException("html");

            return html.StripHtmlTags()
                       .ReplaceCharacterPlaceholders();
        }

        private static string StripHtmlTags(this string html)
        {
            return Regex.Replace(html, "<(/?)([^>]+)>", String.Empty);
        }

        private static string ReplaceCharacterPlaceholders(this string text)
        {
            foreach (var placeholder in characterPlaceholders)
                text = text.Replace(placeholder.Key, placeholder.Value);

            return text;
        }
    }
}

File: C3_WoWFeedReader/FeedGrabberVB/WowFeedGrabber/HtmlConvert.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text.RegularExpressions
Imports System.Runtime.CompilerServices

Namespace WowFeedGrabber
    Public NotInheritable Class HtmlConvert
        Private Shared characterPlaceholders As New Dictionary(Of String, String)

        Shared Sub New()
            characterPlaceholders.Add("&amp;", "&")
            characterPlaceholders.Add("&lt;", "<")
            characterPlaceholders.Add("&gt;", ">")
            characterPlaceholders.Add("&quot;", "'")
            characterPlaceholders.Add("&apos;", "'")


        End Sub


        Public Shared Function ToPlainText(ByVal html As String) As String
            If html Is Nothing Then
                Throw New ArgumentNullException("html")
            End If

            Return ReplaceCharacterPlaceholders(StripHtmlTags(html))
        End Function


        Private Shared Function StripHtmlTags(ByVal html As String) As String
            Return Regex.Replace(html, "<(/?)([^>]+)>", String.Empty)

        End Function

        Private Shared Function ReplaceCharacterPlaceholders(ByVal text As String) As String
            For Each placeholder In characterPlaceholders
                text = text.Replace(placeholder.Key, placeholder.Value)
            Next placeholder

            Return text
        End Function
    End Class
End Namespace
